### PR TITLE
Adding boilerplate

### PR DESCRIPTION
**Pull request by jmfiola**

### CODE REVIEW

File: .github/workflows/quality-checks.yml [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+    steps:
+      - name: Check out source repository
+        uses: actions/checkout@v3
+      - name: Set up Python environment
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.10"
+      - run: pip install -e ".[dev]"
+      - run: ruff check .
+  unit-test-python-310:
```
Comment (alteryx-sezell) [inline .github/workflows/quality-checks.yml]: at some point we should look at tox for doing these. you just give it a list of the python versions to test against. pretty slick.

File: .github/workflows/publish-to-pypi.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      run: |
+        set -x
+        git config --global user.name jmfiola
+        git config --global user.email jmfiola21@gmail.com
+        git config --global pull.rebase false
+        BRANCH=${GITHUB_REF#refs/heads/}
+        pip install --no-cache-dir bump2version
+        git remote add jmfiola https://jmfiola:${{ secrets.PERSONAL_ACCESS_TOKEN}}@github.com/$GITHUB_REPOSITORY
+        git pull --no-edit jmfiola $BRANCH
+        # merge done by git pull does not update submodules changed in $BRANCH
```
Comment (alteryx-sezell) [inline .github/workflows/publish-to-pypi.yml]: this repo isn't using submodules, right? is this cruft left over from other usage?

**Review by alteryx-sezell — APPROVED** [review]
Looks good!

File: .github/workflows/publish-to-pypi.yml (hunk 2 above)
Comment (jmfiola) [reply .github/workflows/publish-to-pypi.yml]: yeah thats just from old stuff. removed!

File: .github/workflows/quality-checks.yml (hunk 1 above)
Comment (jmfiola) [reply .github/workflows/quality-checks.yml]: ooh thats so awesome I had to add it in right now

**Pull request merged**